### PR TITLE
Allow failing due to `git-commit-id-maven-plugin`

### PR DESCRIPTION
**Pull request by eyalkoren**
For failure reproducing purpose

### CODE REVIEW

File: elastic-apm-agent/pom.xml [hunk 1]
```diff
@@ -197,7 +197,7 @@
                         </goals>
                         <configuration>
                             <verbose>true</verbose>
-                            <failOnUnableToExtractRepoInfo>false</failOnUnableToExtractRepoInfo>
+                            <failOnUnableToExtractRepoInfo>true</failOnUnableToExtractRepoInfo>
```
Comment (SylvainJuge) [inline elastic-apm-agent/pom.xml:200]: The default is `true`, thus we can probably remove the config option completely.

File: Jenkinsfile [hunk 2]
```diff
@@ -71,8 +71,8 @@ pipeline {
       steps {
         pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
         deleteDir()
-        gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: true, shallow: false,
-                    reference: '/var/lib/jenkins/.git-references/apm-agent-java.git')
+        // reference repo causes issues while running on Windows with the git-commit-id-maven-plugin
```
Comment (SylvainJuge) [inline Jenkinsfile:74]: [question] the `reference` option seems to be an optimization for faster clone/checkout, is there any noticeable impact here ? If there is one, then maybe using a separate checkout on Windows with the same commit ID could be an option.

File: apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java [hunk 3]
```diff
@@ -97,6 +98,8 @@ public class MockReporter implements Reporter {
 
     private boolean closed;
 
+    private static final long DEFAULT_ASSERTION_TIMEOUT = OS.WINDOWS.isCurrentOs() ? 3000 : 1000;
```
Comment (SylvainJuge) [inline apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java:101]: [minor] just add a comment to help future codebase archeology on the intent here.
```suggestion
    // we have to use a longer timeout on Windows to help reduce flakyness
    private static final long DEFAULT_ASSERTION_TIMEOUT = OS.WINDOWS.isCurrentOs() ? 3000 : 1000;
```

**Review by SylvainJuge — APPROVED** [review]

File: elastic-apm-agent/pom.xml (hunk 1 above)
Comment (eyalkoren) [reply elastic-apm-agent/pom.xml:200]: Yes, but if and when we bump into issues with this plugin again, we won't have to look it up again

**Pull request merged**